framework/src/util/Utils.cpp

```diff
@@ -90,8 +110,35 @@ std::string library_suffix()
 
 }
 
+
 namespace cppmicroservices {
+  //-------------------------------------------------------------------
+  // Unicode Utility functions
+  //-------------------------------------------------------------------
+#ifdef US_PLATFORM_WINDOWS
+
+  // function returns empty string if inStr is empty or if the conversion failed
+  std::wstring ToWString(const std::string& inStr)
+  {
+    int wchar_count = MultiByteToWideChar(CP_UTF8, 0, inStr.c_str(), -1, NULL, 0);
+    std::unique_ptr<wchar_t[]> wBuf(new wchar_t[wchar_count+1]);
```

Comment:
Either the current solution or using a vector instead is fine with me.

But I think we should really differentiate the "invalid utf8" and "empty string" case. Those are too different to be handled in the same way.